compose/aspnet-mssql-compose.md

```diff
@@ -0,0 +1,125 @@
+---
+
+This quick-start guide demonstrates how to use Docker Engine on Linux and Docker Compose to set up and run the sample ASP.NET Core application using the [ASP.NET Core Build image](https://hub.docker.com/r/microsoft/aspnetcore-build/) with the [SQL Server on Linux image](https://hub.docker.com/r/microsoft/mssql-server-linux/). You just need to have [Docker Engine](https://docs.docker.com/engine/installation/) and [Docker Compose](https://docs.docker.com/compose/install/) installed on your platform of choice: Linux, Mac or Windows.
+
+For this sample, we will create a sample `dotnet core web application` using the `aspnetcore-build` Docker image. After that, we will create a `Dockerfile`, configure this app to use our SQL Server database and then create a `docker-compose.yml` that will define the behavior of all of these components.
+
+>**Note**: This sample is made for Docker Engine on Linux. For Windows Containers, please visit [Docker Labs for Windows Containers](https://github.com/docker/labs/tree/master/windows).
+
+# Create the Docker Compose application
+
+1. Create an empty directory named `/app` under the root and navigate into it.
+
+    This directory will be the context of your docker-compose project. For [Docker for Windows](https://docs.docker.com/docker-for-windows/#/shared-drives) and [Docker for Mac](https://docs.docker.com/docker-for-mac/#/file-sharing), you have to set up file sharing for the volume that you need to map. 
+
+1. Within your directory, use the `aspnetcore-build` Docker image to generate a sample web application within the container and into your host machine under the `/app` directory:
+
+        docker run -v /app:/app microsoft/aspnetcore-build:lts /bin/bash -c "cd app && dotnet new -t web"
+
+    >**Note**: If running in Docker for Windows, make sure to add the necessary `C:/` prefix to the host mapped folder and that you are running on Linux containers mode.
+
+1. Create a `Dockerfile` within the `/app` directory and add the following content:
+
+        FROM microsoft/aspnetcore-build:lts
+        COPY . /app
+        WORKDIR /app
+        RUN ["dotnet", "restore"]
+        RUN ["dotnet", "build"]
+        EXPOSE 80/tcp
+        RUN chmod +x ./entrypoint.sh
+        CMD /bin/bash ./entrypoint.sh
+
+    This file defines how to build the web app image. It will use the [microsoft/dotnet](https://hub.docker.com/r/microsoft/dotnet/), map the volume with the generated code, restore the dependencies, build the project and expose port 80. After that, it will call an `entrypoint` script that we will create in the next step.
+
+1. The previous `Dockerfile` makes use of an entrypoint to your webapp Docker image. Create this script in a file called `entrypoint.sh` and paste the contents below.
+
+    >**Note**: Make sure to use UNIX line delimiters. The script won't work if you use Windows-based delimiters (Carriage return and line feed).
+
+        #!/bin/bash
```

Comment:
it's too bad that a custom entrypoint is required. Did you find that this was required to ensure that the database is responding? If it can be done elegantly, I think it'd be preferably to have the user tweak the .NET app to be tolerant of database note responding instead of the wrapper script.